src/commands/init.js

```diff
@@ -815,5 +814,84 @@ const initSubgraphFromContract = async (
     return
   }
 
+  while (addContract) {
+    addContract = await addAnotherContract(toolbox, { protocolInstance, directory })
+  }
+
   printNextSteps(toolbox, { subgraphName, directory }, { commands })
 }
+
+const addAnotherContract = async (toolbox, { protocolInstance, directory }) => {
+  const addContractConfirmation = await toolbox.prompt.confirm('Add another contract?')
+
+  if (addContractConfirmation) {
+    let abiFromFile
+    let ProtocolContract = protocolInstance.getContract()
+  
+    let questions = [
+      {
+        type: 'input',
+        name: 'contract',
+        message: () => `Contract ${ProtocolContract.identifierName()}`,
+        validate: async (value) => {
+          // Validate whether the contract is valid
+          const { valid, error } = validateContract(value, ProtocolContract)
+          return valid ? true : error
+        },
+      },
+      {
+        type: 'select',
+        name: 'localAbi',
+        message: 'Provide local ABI path?',
+        choices: ['yes', 'no'],
+        result: (value) => {
+          abiFromFile = value === 'yes' ? true : false
+          return abiFromFile
+        },
+      },
+      {
+        type: 'input',
+        name: 'abi',
+        message: 'ABI file (path)',
+        skip: () => abiFromFile === false
+      },
+      {
+        type: 'input',
+        name: 'contractName',
+        message: 'Contract Name',
+        initial: 'Contract',
+        validate: (value) => value && value.length > 0,
+      },
+    ]
+  
+    try {
+      let { abi, contract, contractName } = await toolbox.prompt.ask(questions)
+
+      // Get the cwd before process.chdir in order to switch back in the end of command execution
+      const cwd = process.cwd();
+  
+      if (fs.existsSync(directory)) {
+        process.chdir(directory)
+      }
+  
+      let commandLine = ['add', contract, '--contract-name', contractName]
+  
+      if (abiFromFile) {
+        if (abi.includes(directory)) {
+          commandLine.push('--abi', path.normalize(abi.replace(directory, '')))
+        } else {
+          commandLine.push('--abi', abi)
+        }
+      }
+  
+      await graphCli.run(commandLine)
+  
+      process.chdir(cwd)
```

Comment:
Perhaps this should be in the `finally` statement so it will happen no matter if the other lines fail or not.

Reply:
With this change, I think you've forgot to move the `const cwd =` above the `try` block.